go/tdh2/tdh2easy/tdh2easy.go

```diff
@@ -194,22 +194,83 @@ type ciphertextRaw struct {
 	Nonce    []byte
 }
 
+// ciphertextRaw is serialized as _TDH2Ctxt || _SymCtxt || _Nonce
+// where _TDH2Ctxt, _SymCtxt, and _Nonce are length-prefixed byte slices.
+
+func (c ciphertextRaw) Marshal() ([]byte, error) {
+	buf := make([]byte, 0, 4+len(c.TDH2Ctxt)+4+len(c.SymCtxt)+4+len(c.Nonce))
+	buf = append(buf, prefixWithLength(c.TDH2Ctxt)...)
+	buf = append(buf, prefixWithLength(c.SymCtxt)...)
+	buf = append(buf, prefixWithLength(c.Nonce)...)
+	return buf, nil
+}
+
+func (c *ciphertextRaw) Unmarshal(data []byte) error {
+	if len(data) < 4 {
+		return fmt.Errorf("invalid data length")
+	}
+
+	var err error
+	offset := 0
+
+	c.TDH2Ctxt, offset, err = parseLengthPrefixed(data, offset)
+	if err != nil {
+		return fmt.Errorf("cannot decode TDH2 ciphertext: %w", err)
+	}
+	c.SymCtxt, offset, err = parseLengthPrefixed(data, offset)
+	if err != nil {
+		return fmt.Errorf("cannot decode symmetric ciphertext: %w", err)
+	}
+	c.Nonce, _, err = parseLengthPrefixed(data, offset)
+	if err != nil {
+		return fmt.Errorf("cannot decode nonce: %w", err)
+	}
+
+	return nil
+}
+
+// prefixWithLength encodes length-prefixed bytes.
+// The length is encoded as a 4-byte big-endian integer.
+func prefixWithLength(b []byte) []byte {
+	length := len(b)
+	buf := make([]byte, 4+length)
+	binary.BigEndian.PutUint32(buf[:4], uint32(length))
```

Comment:
like before, `length` is 64bit on 64-bit machine so it can overflow here. You need to ensure it is <= MaxUint32 or increase the length prefix